src/Payments.sol

```diff
@@ -0,0 +1,572 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./RateChangeQueue.sol";
+
+interface IArbiter {
+    struct ArbitrationResult {
+        bool approved;
+        uint256 modifiedAmount;
+        uint256 settledUntil;
+    }
+
+    function arbitratePayment(
+        uint256 railId,
+        uint256 proposedAmount,
+        uint256 fromEpoch,
+        uint256 toEpoch
+    ) external returns (ArbitrationResult memory result);
+}
+
+contract Payments is Initializable, UUPSUpgradeable, OwnableUpgradeable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    using RateChangeQueue for RateChangeQueue.Queue;
+
+    struct Account {
+        address ownerAddress;
+        uint256 funds;
+        uint256 lockupCurrent;
+        uint256 lockupRate;
+        uint256 lockupLastSettledAt;
+    }
+
+    struct Rail {
+        bool isActive;
+        address token;
+        address from;
+        address to;
+        address operator;
+        address arbiter;
+        uint256 paymentRate;
+        uint256 lockupPeriod;
+        uint256 lockupFixed;
+        uint256 lastSettledAt;
+    }
+
+    struct OperatorApproval {
+        bool isApproved;
```

Comment:
Do we need this field?

Reply:
@Stebalien Yeah it's the easiest way to track the fact that an operators has been disapproved. Solidity HashMap/Array deletes aren't clean/gas cheap to do.

Reply:
Also, if _operators_ create rails, it makes sense to:

1. Approve the operator with no allowance.
2. Let the operator create rails.
3. Then give them allowance.

(IMO)

Reply:
Have changed `createRail` to automagically approve an operator with zero allowance if it does not already exist.